### PR TITLE
Households have heating system breakdowns, and restricted system choices at breakdown

### PR DESCRIPTION
**Pull request by stephanie-chan-cnz**
- Initialise households with `heating_system_install_date`, to replace household `heating_system_age`. This avoids us needing to update heating system age at each timestep.
- Add household method to evaluate if heating is functioning, based on age of heating system and Weibull hazard function parameters
- Modify `household.get_heating_system_options` so heat pumps are not options during a breakdown (due to high industry lead times), unless a household already has a heat pump installed. In that event, the specific type of heat pump already installed (air/ground source) remains an option.

### CODE REVIEW

File: simulation/agents.py [hunk 1]
```diff
@@ -50,6 +54,7 @@ def __init__(
         property_type: PropertyType,
         built_form: BuiltForm,
         heating_system: HeatingSystem,
+        heating_system_install_date: datetime.datetime,
```
Comment (tomwphillips) [inline simulation/agents.py:57]: Why is `household.heating_system_install_date` a `datetime` and not a `date`?

File: simulation/tests/test_agents.py [hunk 2]
```diff
@@ -241,42 +246,106 @@ def test_households_not_suitable_archetype_are_not_heat_pump_suitable(
...
+        event_trigger,
     ) -> None:
 
         unaware_household = household_factory(is_heat_pump_aware=False)
         model = model_factory()
         assert not HEAT_PUMPS.intersection(
-            unaware_household.get_heating_system_options(model)
+            unaware_household.get_heating_system_options(
+                model, event_trigger=event_trigger
+            )
```
Comment (tomwphillips) [inline simulation/tests/test_agents.py:274]: I can see it's not something you added in this PR, so don't feel obliged to change it, but I had to think about what's going on here with `assert not` and a set. I think something more explicit like the following is clearer.

```
heating_system_options = unaware_household.get_heating_system_options(model, event_trigger)
assert heating_system_options.intersection(HEAT_PUMPS) == {}
```

File: simulation/tests/test_agents.py [hunk 3]
```diff
@@ -241,42 +246,106 @@ def test_households_not_suitable_archetype_are_not_heat_pump_suitable(
...
-        self,
+        self, event_trigger
     ) -> None:
 
         off_gas_grid_household = household_factory(off_gas_grid=True)
         model = model_factory()
         assert not {HeatingSystem.BOILER_GAS}.intersection(
-            off_gas_grid_household.get_heating_system_options(model)
+            off_gas_grid_household.get_heating_system_options(
+                model, event_trigger=event_trigger
```
Comment (tomwphillips) [inline simulation/tests/test_agents.py:286]: Again, not obligatory, but no reason for the keyword argument with `event_trigger` in these tests.

**Review by tomwphillips — COMMENTED** [review]
Just a question about the `datetime` and a couple of other optional things.

File: simulation/agents.py (hunk 1 above)
Comment (stephanie-chan-cnz) [reply simulation/agents.py:57]: True, I'll change it to a date, no need to have a lifetime in microseconds!

**Review by tomwphillips — APPROVED** [review]

**Pull request merged**